### PR TITLE
resource/aws_db_instance: Adds support for domain joining RDS instances

### PR DESCRIPTION
**Pull request by mburtless**
Adds domain and ~domain-iam-role-name~ domain_iam_role_name parameters to resource aws_db_instance.  Based on work from hashicorp/terraform#14425 and #1367.

As #1367 has been closed, I've merged commits from branch by @mikewalker125 and added acceptance tests and updated documentation.

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSDBInstance_MSSQL_Domain'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSDBInstance_MSSQL_Domain -timeout 120m
=== RUN   TestAccAWSDBInstance_MSSQL_Domain

--- PASS: TestAccAWSDBInstance_MSSQL_Domain (2516.52s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       2517.593s
```

### CODE REVIEW

File: aws/resource_aws_db_instance.go [hunk 1]
```diff
@@ -404,6 +404,18 @@ func resourceAwsDbInstance() *schema.Resource {
 				},
 			},
 
+			"domain": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```
[the remark this answers is not in the document]
Comment (mburtless) [reply aws/resource_aws_db_instance.go]: Not sure what the original reason was as this was merged from the previous code for this feature committed by @mikewalker125 in #1367.  That said, your feedback makes sense to me.  I'll remove `Computed: true`

File: aws/resource_aws_db_instance.go [hunk 2]
```diff
@@ -404,6 +404,18 @@ func resourceAwsDbInstance() *schema.Resource {
...
+			},
+
+			"domain_iam_role_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```
[the remark this answers is not in the document]
Comment (mburtless) [reply aws/resource_aws_db_instance.go]: Same as above, this was merged from the previous code for this feature committed by @mikewalker125 in #1367.  Will remove here as well.

File: aws/resource_aws_db_instance_test.go [hunk 3]
```diff
@@ -571,6 +571,33 @@ func TestAccAWSDBInstance_MSSQL_TZ(t *testing.T) {
...
+						"aws_db_instance.mssql", "domain"),
+					resource.TestCheckResourceAttrSet(
+						"aws_db_instance.mssql", "domain_iam_role_name"),
+					resource.TestCheckResourceAttr(
+						"aws_db_instance.mssql", "allocated_storage", "20"),
+					resource.TestCheckResourceAttr(
+						"aws_db_instance.mssql", "engine", "sqlserver-ex"),
+				),
+			},
+		},
```
[the remark this answers is not in the document]
Comment (mburtless) [reply aws/resource_aws_db_instance_test.go]: Makes sense, will add a step to test removing the instance from the domain.

**Comment by mburtless** [comment]
@bflad - Thanks for the feedback.  All of those comments make sense and I should be able to get the branch updated in the next few days.

Regarding add support for `Domain` and `DomainIAMRole` in Terraform's use of `RestoreDBInstanceFromDBSnapshotInput`, `CreateDBInstanceReadReplicaInput` and `RestoreDBInstanceFromS3Input`; is your preference that we include that in this PR or a future one?

**Comment by bflad** [comment]
We should either include it now or document that it currently takes two `terraform apply`s in the attribute documentation in those situations. 👍  Inevitably without at least documenting it, there will be bug report(s) submitted for the unexpected behavior.

**Comment by mburtless** [comment]
Noting that after doing some digging, it doesn't appear as if `CreateDBInstanceReadReplicaInput` supports the `domain` or `domain_iam_role_name` arguments, as RDS instances using the SQL Server engine do not currently support read replicas.

Source:  [AWS](https://aws.amazon.com/rds/details/read-replicas/) mentions "Read replicas are available in Amazon RDS for MySQL, MariaDB, and PostgreSQL as well as Amazon Aurora."

Additionally the [CreateDBInstanceReadReplicaInput](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#CreateDBInstanceReadReplicaInput) struct does not have fields or setters for Domain or DomainIAMRoleName.

Will continue to explore implementation of these arguments for `RestoreDBInstanceFromDBSnapshotInput` and `RestoreDBInstanceFromS3Input`

@bflad - Was there another use case you had in mind when you mentioned `CreateDBInstanceReadReplicaInput` above?

**Comment by mburtless** [comment]
@bflad - According to [AWS Go API Reference](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#RestoreDBInstanceFromS3Input)  `RestoreDBInstanceFromS3Input` also does not support `Domain` or `DomainIAMRoleName` fields.

With that in mind, I think all the features we need to support have been committed to this PR and are ready for another review (support for `RestoreDBInstanceFromDBSnapshotInput` added in last commit), whenever you get a chance.

**Comment by mburtless** [comment]
Great news, glad I could help out!  Thanks for all the help and feedback here Brian, and to @mikewalker125 for his work on the previous PR.

**Comment by bflad** [comment]
This has been released in [version 1.36.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1360-september-13-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by RaviKumar1209** [comment]
can we now use terraform to add RDS instances to domain?

**Pull request merged**